stablehlo/reference/Ops.h

```diff
@@ -46,8 +49,18 @@ Tensor evalSubtractOp(const Tensor &lhs, const Tensor &rhs, Type resultType);
 Tensor evalTanhOp(const Tensor &operand, Type resultType);
 Tensor evalTransposeOp(const Tensor &operand, ArrayRef<int64_t> permutation,
                        Type resultType);
+SmallVector<Tensor> evalWhileOp(ArrayRef<Tensor> inputs, Region &cond,
+                                Region &body, const Scope &scope);
```

Comment:
Why does the scope have to be non-null here? Theoretically, nothing should stop us from calling this function with a nullptr scope if the condition and the body of the loop don't capture anything from above?

Reply:
>> Why does the scope have to be non-null here?


This `scope`, created at `eval(Region..` and  passed into the `evalWhileOp`,   refers to the scope where the while operation is defined and IMO it should never be null.

>>  Theoretically, nothing should stop us from calling this function with a nullptr scope if the condition and the body of the loop don't capture anything from above?

If the  `cond` and `body` regions do not capture any thing from above, that would be realized using the scope object for those regions to bear a parent scope = `nullptr`. 

 Sorry if I misunderstood this question. Please let me know.

Reply:
I agree that we will not encounter a non-null `scope` during normal execution of the `eval` function. From the perspective of that use case, the current signature is just fine.

However, I also noticed that the non-nullness of the `scope` is not used in `evalWhileOp`, and this got me thinking about alternative use cases. Imagine if someone wants to just evaluate a WhileOp, without worrying about the enclosing function. If the scope is nullable, they can easily do that, but if the scope is not nullable, they will have to create a dummy scope.

Personally, I don't have a strong preference either way - just wanted to elaborate on where I was coming from. Please feel free to resolve how you see fit.